src/FakeItEasy/Configuration/ArgumentCollection.cs

```diff
@@ -68,6 +69,8 @@ public int Count
         /// </summary>
         public IEnumerable<string> ArgumentNames { get; }
 
+        internal MethodInfo Method { get; }

```

Comment:
Yeah, I'm not thrilled about it. But it's needed to verify the delegates signatures, and I didn't see a better way to do it. What bothers me a little is that `ArgumentCollection` has two constructors, one of which doesn't take a `MethodInfo`. It's only used in unit tests, but we could easily end up with a null method if we're not careful.